File: backend/app/models/system_log.py
from sqlalchemy import Column, Integer, String, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.models.base import Base


class SystemLog(Base):
    __tablename__ = "system_logs"

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    action = Column(String(20), nullable=False)
    ip_address = Column(String(45))
    user_agent = Column(Text)
    device_type = Column(String(20))
    browser = Column(String(50))
    os = Column(String(50))
    device_name = Column(String(100))
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    user = relationship("User", backref="system_logs")

File: whisper-app/whisper_app/lib/features/auth/data/models/system_log.dart
class SystemLog {
  final int id;
  final int userId;
  final String action;
  final String? ipAddress;
  final String? userAgent;
  final String? deviceType;
  final String? browser;
  final String? os;
  final String? deviceName;
  final DateTime createdAt;

  SystemLog({
    required this.id,
    required this.userId,
    required this.action,
    this.ipAddress,
    this.userAgent,
    this.deviceType,
    this.browser,
    this.os,
    this.deviceName,
    required this.createdAt,
  });

  factory SystemLog.fromJson(Map<String, dynamic> json) {
    return SystemLog(
      id: json['id'],
      userId: json['user_id'],
      action: json['action'],
      ipAddress: json['ip_address'],
      userAgent: json['user_agent'],
      deviceType: json['device_type'],
      browser: json['browser'],
      os: json['os'],
      deviceName: json['device_name'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}